index.js

```diff
@@ -22,57 +19,85 @@ var hmac = sodium.crypto_auth
 
 exports.hash = u.hash
 
-exports.getTag = u.getTag
+exports.getFeedType = u.getFeedType
+exports.getTag = u.getFeedType // deprecated
 
 function isObject (o) {
   return 'object' === typeof o
 }
 
-function isFunction (f) {
-  return 'function' === typeof f
-}
-
 function isString(s) {
   return 'string' === typeof s
 }
 
-var curves = {}
-curves.ed25519 = require('./sodium')
+const feedTypes = {
+  ed25519: require('./sodium'),
+  'ed25519.test': require('./sodium')
+}
+
+exports.use = (name, object) => {
+  if (typeof name !== 'string' || name.length === 0) {
+    throw new Error(`Invalid name: "${name}", expected string with non-zero length`)
+  }
+
+  const requiredMethods = [
+    'generate',
+    'sign',
+    'verify'
+  ]
+
+  const isNotObject = typeof object !== 'object'
+  const isInvalidObject = isNotObject || requiredMethods.some(methodName => 
+    typeof object[methodName] !== 'function'
+  )
+
+  if (isInvalidObject) {
+    const expectedMethods = requiredMethods.join(', ')
+    console.log(object)
+    throw new Error(`Invalid object. Missing required methods, expected: ${expectedMethods}`)
+  }
+
+  if (feedTypes[name] != null) {
+    throw new Error(`Duplicate feed type: "${name}"`)
+  }
+
+  feedTypes[name] = object
+}
 
-function getCurve(keys) {
-  var curve = keys.curve
+function getFeedType(keys) {
+  let { feedType } = keys
+  feedType = feedType || keys.curve
```

Comment:
<style> If you're using es6 tools do this instead:

```js
const {
  feedType = keys.curve
} = keys
```

actually this is a mess (it works well for destructuring multiple opts with defaults)
This is clearer surely

```js
const feedType = keys.feedType || keys.curve
```